File: app/src/main/java/com/micewine/emu/fragments/RatManagerFragment.java
package com.micewine.emu.fragments;

import static com.micewine.emu.core.RatPackageManager.listRatPackages;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.RecyclerView;

import com.micewine.emu.R;
import com.micewine.emu.adapters.AdapterRatPackage;
import com.micewine.emu.core.RatPackageManager;

import java.util.ArrayList;
import java.util.List;

public class RatManagerFragment extends Fragment {
    private final String prefix;
    private final String anotherPrefix;
    private final int type;

    public RatManagerFragment(String prefix, int type) {
        this.prefix = prefix;
        this.type = type;
        this.anotherPrefix = prefix;
    }

    public RatManagerFragment(String prefix, int type, String anotherPrefix) {
        this.prefix = prefix;
        this.type = type;
        this.anotherPrefix = anotherPrefix;
    }

    private RecyclerView recyclerView;
    private final ArrayList<AdapterRatPackage.Item> ratList = new ArrayList<>();

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View rootView = inflater.inflate(R.layout.fragment_general_settings, container, false);

        recyclerView = rootView.findViewById(R.id.recyclerViewGeneralSettings);

        setAdapter();

        return rootView;
    }

    private void setAdapter() {
        recyclerView.setAdapter(new AdapterRatPackage(ratList, requireActivity(), false));

        ratList.clear();

        List<RatPackageManager.RatPackage> ratPackageList = listRatPackages(prefix, anotherPrefix);

        for (RatPackageManager.RatPackage ratPackage : ratPackageList) {
            addToAdapter(ratPackage.getName(), ratPackage.getVersion(), ratPackage.getFolderName(), ratPackage.getIsUserInstalled());
        }
    }

    private void addToAdapter(String title, String description, String itemFolderId, boolean canDelete) {
        ratList.add(
                new AdapterRatPackage.Item(title, description, itemFolderId, type, canDelete, null, false)
        );
    }
}

File: app/src/main/java/com/micewine/emu/fragments/RatManagerFragment.kt
package com.micewine.emu.fragments

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.RecyclerView
import com.micewine.emu.R
import com.micewine.emu.adapters.AdapterRatPackage
import com.micewine.emu.core.RatPackageManager.listRatPackages

class RatManagerFragment(private val prefix: String, private val type: Int, private val anotherPrefix: String = prefix) : Fragment() {
    private val ratList: MutableList<AdapterRatPackage.Item> = mutableListOf()
    private var rootView: View? = null
    private var recyclerView: RecyclerView? = null

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        rootView = inflater.inflate(R.layout.fragment_general_settings, container, false)
        recyclerView = rootView?.findViewById(R.id.recyclerViewGeneralSettings)

        setAdapter()

        return rootView
    }

    private fun setAdapter() {
        recyclerView?.setAdapter(
            AdapterRatPackage(ratList, requireContext())
        )

        ratList.clear()

        listRatPackages(prefix, anotherPrefix).forEach {
            addToAdapter(it.name!!, it.version!!, it.folderName!!, it.isUserInstalled!!)
        }
    }

    private fun addToAdapter(title: String, description: String, driverFolderId: String, canDelete: Boolean) {
        ratList.add(
            AdapterRatPackage.Item(title, description, driverFolderId, type, canDelete)
        )
    }
}
